src/lib/pubkey/kyber/kyber_common/kyber_symmetric_primitives.h

```diff
@@ -95,8 +104,12 @@ class Kyber_Symmetric_Primitives {
       }
 
    protected:
+      virtual std::optional<std::array<uint8_t, 1>> seed_expansion_domain_separator(
```

Comment:
Why the `array` and not just `std::optional<uint8_t>`?

Reply:
Oh maybe this just makes life easier wrt `concat`?

Reply:
Yep. That's the reason. I did look into extending concat to take scalar values along with ranges but didn't find an easy enough way to allow that.

We might want to look into that later though.